frontend/pages/LabelPage/LabelPage.tsx

```diff
@@ -0,0 +1,205 @@
+
+import PATHS from "router/paths";
+import MainContent from "components/MainContent";
+import SidePanelContent from "components/SidePanelContent";
+import QuerySidePanel from "components/side_panels/QuerySidePanel";
+import { QueryContext } from "context/query";
+import { NotificationContext } from "context/notification";
+import { IApiError } from "interfaces/errors";
+import { ILabel, ILabelFormData } from "interfaces/label";
+import labelsAPI, { ILabelsResponse } from "services/entities/labels";
+import deepDifference from "utilities/deep_difference";
+import LabelForm from "./LabelForm";
+
+const baseClass = "label-page";
+
+interface ILabelPageProps {
+  router: InjectedRouter;
+  params: Params;
+  location: {
+    pathname: string;
+  };
+}
+
+const DEFAULT_CREATE_LABEL_ERRORS = {
+  name: "",
+};
+
+const LabelPage = ({
+  router,
+  params,
+  location,
+}: ILabelPageProps): JSX.Element | null => {
+  const isEditLabel = !location.pathname.includes("new");
+
+  const [selectedLabel, setSelectedLabel] = useState<ILabel>();
+  const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(true);
+  const [
+    showOpenSchemaActionText,
+    setShowOpenSchemaActionText,
+  ] = useState<boolean>(false);
+  const [labelValidator, setLabelValidator] = useState<{
+    [key: string]: string;
+  }>(DEFAULT_CREATE_LABEL_ERRORS);
+  const { selectedOsqueryTable, setSelectedOsqueryTable } = useContext(
+    QueryContext
+  );
+
+  const { renderFlash } = useContext(NotificationContext);
+
+  const showSidebar = isSidebarOpen;
+
+  const { data: labels, error: labelsError } = useQuery<
+    ILabelsResponse,
+    Error,
+    ILabel[]
+  >(["labels"], () => labelsAPI.loadAll(), {
+    select: (data: ILabelsResponse) => data.labels,
+    onSuccess: (responseLabels: ILabel[]) => {
+      if (params.label_id) {
+        const selectLabel = responseLabels.find(
+          (label) => label.id === parseInt(params.label_id, 10)
+        );
+        setSelectedLabel(selectLabel);
+      }
+    },
+  });
+
+  useEffect(() => {
+    setShowOpenSchemaActionText(!isSidebarOpen);
+  }, [isSidebarOpen]);
+
+  const onCloseSchemaSidebar = () => {
+    setIsSidebarOpen(false);
+  };
+
+  const onOpenSchemaSidebar = () => {
+    setIsSidebarOpen(true);
+  };
+
+  const onOsqueryTableSelect = (tableName: string) => {
+    setSelectedOsqueryTable(tableName);
+  };
+
+  const onEditLabel = (formData: ILabelFormData) => {
+    if (!selectedLabel) {
+      console.error("Label isn't available. This should not happen.");
+      return;
+    }
+
+    const updateAttrs = deepDifference(formData, selectedLabel);
+
+    labelsAPI
+      .update(selectedLabel, updateAttrs)
+      .then(() => {
+        router.push(PATHS.MANAGE_HOSTS_LABEL(selectedLabel.id));
+        renderFlash(
+          "success",
+          "Label updated. Try refreshing this page in just a moment to see the updated host count for your label."
+        );
+        setLabelValidator({});
```

Comment:
at this point this component is no longer rendered on the screen so we shouldnt need to reset the `labelValidator` object

Reply:
Done